clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -779,7 +787,10 @@ protected void close(Timer timer) {
             // Synchronize after closing the heartbeat thread since heartbeat thread
             // needs this lock to complete and terminate after close flag is set.
             synchronized (this) {
-                if (leaveGroupOnClose) {
+                // Starting from 2.2, only dynamic members will send LeaveGroupRequest to the broker,
+                // consumer with valid group.instance.id is viewed as static member that never sends LeaveGroup,
+                // and the membership expiration is only controlled by session timeout.
+                if (this.groupInstanceId.equals(JoinGroupRequest.UNKNOWN_GROUP_INSTANCE_ID)) {
```

Comment:
`leaveGroupOnClose` was set through internal config `internal.leave.group.on.close` which is by default set to true for normal consumer/connect, but set to false for streams. Checking groupInstanceId could perfectly remove this internal config without backward compatibility concern (since the config is not exposed)